pkg/kubelet/kubeletconfig/util/files/files.go

```diff
@@ -83,6 +83,10 @@ func ReplaceFile(fs utilfs.Filesystem, path string, data []byte) error {
 	if _, err := tmpFile.Write(data); err != nil {
 		return err
 	}
+	// sync file, to ensure it's written in case a hard reset happens
+	if err := tmpFile.Sync(); err != nil {
+		return err
+	}
 	if err := tmpFile.Close(); err != nil {
```

Comment:
This looks like a (pre-existing) file leak to me.
I.e. if `Write()` or `Sync()` return an error or something panics then the file will not be closed properly. See https://joeshaw.org/dont-defer-close-on-writable-files/ for a good example of how to handle that with `defer` and namer return values.

Another thing that comes to mind is deleting the temporary file (after closing it) if there was an error. Otherwise it is left behind and just consumes space.

Reply:
Ah, thanks! I'll send a fix.